src/components/BreedsList.vue

```diff
@@ -0,0 +1,75 @@
+<template>
+  <section>
+    <p v-if="isLoading">The Data is Loading ...</p>
+    <p v-else-if="!isLoading && error">{{ error }}</p>
+    <p v-else-if="isTimeOut && breedsList.length === 0">
+      The request takes longer than expected...
+    </p>
+    <p v-else-if="!isLoading && breedsList.length === 0">
+      No Data Found. Please try again later.
+    </p>
+    <div v-else>
+      <div v-for="breed in breedsList">
+        <h3>{{ breed.name }}</h3>
+        <span>{{ breed.breed_group }}</span>
+        <span>{{ breed.life_span }}</span>
+        <span>{{ breed.height.metric }}cm</span>
+        <span>{{ breed.weight.metric }}kg</span>
+        <img
+          class="breedsList_image"
+          :src="breed.image.url"
+          :alt="`Picture of the ${breed.name} dog`"
+        />
+      </div>
+    </div>
+  </section>
+</template>
+
+<script>
+import { fetchBreeds } from "../data.js";
+export default {
+  data() {
+    return {
+      breedsList: [],
+      isLoading: false,
+      error: null,
+      isTimeOut: false,
+    };
+  },
+  methods: {
+    fetchData() {
+      this.isLoading = true;
+      setTimeout(() => {
+        if (this.isLoading) {
+          this.isTimeOut = true;
+          this.isLoading = false;
+        }
+      }, 3000);
+      fetchBreeds()
+        .then((results) => {
+          this.isLoading = false;
+          if (results.length > 0) {
```

Comment:
to lahko poštimaš še v naslednjem PRju.. ker tukej nastaviš breeds na result, če ni prazen array, če pa je prazen array pa nastaviš na prazen array.. 

`this.breedsList = results;` ta ena vrstica je čist isto kot if else pri tebi, ane?